.eslintrc.js

```diff
@@ -2,7 +2,26 @@
  * @type {import('eslint').Linter.Config}
  */
 const config = {
-  extends: ['algolia', 'algolia/jest', 'algolia/react', 'algolia/typescript'],
+  extends: [
+    'algolia',
+    'algolia/jest',
+    'algolia/react',
+    'algolia/typescript',
+    'plugin:react-hooks/recommended',
+  ],
+  settings: {
+    react: {
+      version: 'detect',
+    },
+    'import/resolver': {
+      node: {
+        // The migration is an incremental process so we import TypeScript modules
+        // from JavaScript files.
+        // By default, `import/resolver` only supports JavaScript modules.
+        extensions: ['.js', '.ts', '.tsx'],
+      },
+    },
+  },
   rules: {
```

Comment:
I think we should comment rules that we disable to clarify the rationale for future readers.

This also lets us keep an eye on what we only disabled because we're in a migration process (and need to reenable later) and what constitutes an actual decision.

Reply:
I agree. I added a few comments.